lib/OpenQA/Worker.pm

```diff
@@ -67,12 +67,14 @@ sub main {
             $shared_cache = prepare_cache_directory($h, $worker_settings->{CACHEDIRECTORY});
         }
         # this is being also duplicated by OpenQA::Test::Utils since 49c06362d
-        my @dirs = ($host_settings->{$h}{SHARE_DIRECTORY}, catdir($OpenQA::Utils::prjdir, 'share'));
-        ($dir) = grep { $_ && -d } @dirs;
-        unless ($dir) {
-            map { log_debug("Found possible working directory for $h: $_") if $_ } @dirs;
-            log_error("Ignoring host '$h': Working directory does not exist.");
-            next;
+        if ($worker_settings->{SHARE_DIRECTORY}) {
+            my @dirs = ($host_settings->{$h}{SHARE_DIRECTORY}, catdir($OpenQA::Utils::prjdir, 'share'));
```

Comment:
Sorry, but that is just wrong.

The code tries to find a share directory from either setting or hardcoded 'share'. If you make this conditional, you can basically remove the whole codeblock.